File: Javascript/Intersection-Of-Two-Arrays.js
/**
 * @param {number[]} nums1
 * @param {number[]} nums2
 * @return {number[]}
 */
var intersection = function (nums1, nums2) {
  const numsMap = new Set(nums1);
  const intersectionOfNums = [];

  for (let num of nums2) {
    if (numsMap.has(num)) {
      intersectionOfNums.push(num);
      numsMap.delete(num);
    }
  }

  return intersectionOfNums;
};

File: Rust/Intersection-Of-Two-Arrays.rs
impl Solution {
    pub fn intersection(nums1: Vec<i32>, nums2: Vec<i32>) -> Vec<i32> {
        let mut nums_map = HashSet::new();
        let mut intersection_of_nums = Vec::new();

        for num in nums1 {
            nums_map.insert(num);
        }

        for num in nums2 {
            if nums_map.contains(&num) {
                intersection_of_nums.push(num);
                nums_map.remove(&num);
            }
        }

        intersection_of_nums
    }
}
